enumerable_methods.rb

```diff
@@ -0,0 +1,159 @@
+# frozen_string_literal: true
+
+module Enumerable # rubocop:disable Metrics/ModuleLength
+  # my_each
+  def my_each
+    return to_enum unless block_given?
+
+    i = 0
+    while i < size
+      yield(self[i])
+      i += 1
+    end
+  end
+
+  # my_each_with_index
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    i = 0
+    while i < size
+      yield(self[i], i)
+      i += 1
+    end
+  end
+
+  # my_select
+  def my_select
+    return to_enum unless block_given?
+
+    arr = []
+    my_each do |a|
+      arr.push(a) if yield(a)
+    end
+    arr
+  end
+
+  # my_all
+  def my_all?(arg = nil) # rubocop:disable Layout/TrailingWhitespace
+    if block_given?
+      my_each { |a| return false unless yield(a) }
+      return true
+    end
+    if arg.class == Regexp
+      my_each { |a| return false unless a.match(arg) }
+      return true
+    elsif arg.class == Class
+      my_each { |a| return false unless a.class == arg || a.class.superclass == arg }
+      return true
+    elsif arg.nil? && !block_given? && !empty?
+      return false
+    end
+    true
+  end
+
+  # my_any
+  def my_any?(arg = nil)
+    if block_given?
+      my_each { |a| return true if yield(a) }
+      return false
+    end
+    if arg.class == Regexp
+      my_each { |a| return true if a.match(arg) }
+      return false
+    elsif arg.class == Class
+      my_each { |a| return true if a.class == arg || a.class.superclass == arg }
+      return false
+    elsif arg.nil? && !block_given? && !empty?
+      return true
+    end
+    false
+  end
+
+  # my_none
+  def my_none?(arg = nil)
+    if block_given?
+      my_each { |a| return false if yield(a) }
+      return true
+    end
+    if arg.class == Regexp
+      my_each { |a| return false if a.match(arg) }
+      return true
+    elsif arg.class == Class
+      my_each { |a| return true unless a.class == arg || !a.class.superclass == arg }
+      return true
+    elsif arg.nil? && !block_given? && !empty?
+      return true
+    end
+    true
+  end
```

Comment:
**[REQUIRED]**
* When no block or argument is given returns true **only if** none of the collection members is true.
![none2](https://user-images.githubusercontent.com/44072711/68568711-5f2acf00-042a-11ea-9a96-368b34dc789b.png).
* When a class is passed as an argument returns true **if none** of the collection is a member of such class.
![noe3](https://user-images.githubusercontent.com/44072711/68569165-6dc5b600-042b-11ea-8e0d-14df2949023e.png)

* when a pattern other than Regex or a Class is given returns true only **if none** of the collection matches the pattern.
![noe4](https://user-images.githubusercontent.com/44072711/68569118-54246e80-042b-11ea-9e7d-4d0cfd8db3aa.png)